### PR TITLE
init dev branch and update version infos

### PR DESCRIPTION
**Pull request by buhtz**
Just minor modifications.

 - Introducing the `dev` branch based on the _gitflow_ branching model. Should be used for Pull Requests and nearly all other activities. The commits in `main` should reflect releases only; exceptions possible of course. 
 - Update version number to 1.3.4-dev
 - Mention the `dev` branch in README contributors section.

### CODE REVIEW

File: common/man/C/backintime-config.1 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.TH backintime-config 1 "Nov 2022" "version 1.3.3" "USER COMMANDS"
+.TH backintime-config 1 "Nov 2022" "version 1.3.4-dev" "USER COMMANDS"
```
Comment (aryoda) [inline common/man/C/backintime-config.1:1]: "Nov 2022" should probably be "Jan 2023" now?

File: common/man/C/backintime.1 [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.TH backintime 1 "Oct 2022" "version 1.3.3" "USER COMMANDS"
+.TH backintime 1 "Oct 2022" "version 1.3.4-dev" "USER COMMANDS"
```
Comment (aryoda) [inline common/man/C/backintime.1:1]: "Oct 2022" should probably be "Jan 2023" now?

File: qt/man/C/backintime-qt.1 [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.TH backintime-qt 1 "Oct 2022" "version 1.3.3" "USER COMMANDS"
+.TH backintime-qt 1 "Oct 2022" "version 1.3.4-dev" "USER COMMANDS"
```
Comment (aryoda) [inline qt/man/C/backintime-qt.1:1]: "Oct 2022" should probably be "Jan 2023" now?

**Review by aryoda — COMMENTED** [review]
THX a lot for all the work you have done to introduce a new and better branching model!
I have just a few minor notes on the version date of the man pages (should probably be updated too)

**Comment by buhtz** [comment]
Dear Jürgen,
thanks for your feedback.

Good note about the date. I will note this on my how-to-release-checklist.

**Review by aryoda — APPROVED** [review]

**Pull request merged**